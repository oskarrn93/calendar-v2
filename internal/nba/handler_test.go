package nba_test

import (
	"context"
	"fmt"
	"io"
	"log"
	"log/slog"
	"testing"

	"github.com/oskarrn93/calendar-v2/internal/logging"
	"github.com/oskarrn93/calendar-v2/internal/nba"
	"github.com/oskarrn93/calendar-v2/internal/rapidapi"
	"github.com/oskarrn93/calendar-v2/internal/testdata"
	"github.com/oskarrn93/calendar-v2/internal/testutil"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/go-resty/resty/v2"
	"github.com/jarcoal/httpmock"
	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/mock"
	"github.com/stretchr/testify/require"
)

func readGamesTestData() []byte {
	// Use saved api response so we don't need to make an external request

	jsonFile, err := testdata.Content.Open("nba/games/celtics.json")
	if err != nil {
		log.Fatal("Failed to open games test data", err)
	}
	defer jsonFile.Close()

	data, err := io.ReadAll(jsonFile)
	if err != nil {
		log.Fatal("Failed to read games test data", err)
	}

	return data
}

type MockStorage struct {
	mock.Mock
}

func (m *MockStorage) Upload(ctx context.Context, filename string, data []byte, logger *slog.Logger) error {
	args := m.Called(ctx, filename, data, logger)
	return args.Error(0)
}

func TestGetGames(t *testing.T) {
	//Arrange
	httpmock.Activate()
	defer httpmock.DeactivateAndReset()

	httpClient := resty.New()
	httpmock.ActivateNonDefault(httpClient.GetClient())

	mockConfig := testutil.GetMockAppConfig()

	rapidApi := rapidapi.New(httpClient, mockConfig.RapidApi)
	mockStorage := MockStorage{}

	nbaHandler := nba.NewHandler(rapidApi, &mockStorage, logging.New())

	gamesTestData := string(readGamesTestData())

	// Mock http request
	expectedUrl := fmt.Sprintf("%s/games?season=%d&team=%d", mockConfig.RapidApi.NBA.BaseUrl, mockConfig.RapidApi.NBA.Season, nba.CELTICS_TEAM_ID)
	httpmock.RegisterResponder("GET", expectedUrl,
		httpmock.NewStringResponder(200, gamesTestData))

	// Act
	result, err := nbaHandler.GetGames([]nba.TeamID{nba.CELTICS_TEAM_ID})
	require.NoError(t, err)

	// Assert
	assert.Len(t, result, 87)

	for _, game := range result {
		if game.Teams.Home.Id != int(nba.CELTICS_TEAM_ID) && game.Teams.Visitors.Id != int(nba.CELTICS_TEAM_ID) {
			t.Errorf("Expected either home or visitor team to be CELTICS_TEAM_ID, but got home: %d, visitor: %d", game.Teams.Home.Id, game.Teams.Visitors.Id)
		}
	}

	snaps.MatchSnapshot(t, result)
}
