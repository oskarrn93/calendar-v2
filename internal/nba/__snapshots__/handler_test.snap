
[TestGetGames - 1]
[]nba.Game{
    {
        Id:   12482,
        Date: nba.GameDate{
            Start: time.Date(2023, time.October, 8, 22, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:27, Name:"Philadelphia 76ers"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12486,
        Date: nba.GameDate{
            Start: time.Date(2023, time.October, 9, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:24, Name:"New York Knicks"},
        },
    },
    {
        Id:   12498,
        Date: nba.GameDate{
            Start: time.Date(2023, time.October, 11, 23, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:27, Name:"Philadelphia 76ers"},
        },
    },
    {
        Id:   12526,
        Date: nba.GameDate{
            Start: time.Date(2023, time.October, 17, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:24, Name:"New York Knicks"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12535,
        Date: nba.GameDate{
            Start: time.Date(2023, time.October, 19, 23, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:5, Name:"Charlotte Hornets"},
        },
    },
    {
        Id:   12553,
        Date: nba.GameDate{
            Start: time.Date(2023, time.October, 25, 23, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:24, Name:"New York Knicks"},
        },
    },
    {
        Id:   12568,
        Date: nba.GameDate{
            Start: time.Date(2023, time.October, 27, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:20, Name:"Miami Heat"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12591,
        Date: nba.GameDate{
            Start: time.Date(2023, time.October, 30, 23, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:41, Name:"Washington Wizards"},
        },
    },
    {
        Id:   12606,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 1, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:15, Name:"Indiana Pacers"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12631,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 5, 0, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:4, Name:"Brooklyn Nets"},
        },
    },
    {
        Id:   12649,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 7, 1, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:22, Name:"Minnesota Timberwolves"},
        },
    },
    {
        Id:   12654,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 9, 0, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:27, Name:"Philadelphia 76ers"},
        },
    },
    {
        Id:   12670,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 11, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:4, Name:"Brooklyn Nets"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12678,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 12, 0, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:38, Name:"Toronto Raptors"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12692,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 14, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:24, Name:"New York Knicks"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12708,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 16, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:27, Name:"Philadelphia 76ers"},
        },
    },
    {
        Id:   12720,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 18, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:38, Name:"Toronto Raptors"},
        },
    },
    {
        Id:   12738,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 20, 1, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:19, Name:"Memphis Grizzlies"},
        },
    },
    {
        Id:   12742,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 21, 0, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:5, Name:"Charlotte Hornets"},
        },
    },
    {
        Id:   12758,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 23, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:21, Name:"Milwaukee Bucks"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12769,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 24, 19, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:26, Name:"Orlando Magic"},
        },
    },
    {
        Id:   12786,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 26, 23, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:1, Name:"Atlanta Hawks"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12798,
        Date: nba.GameDate{
            Start: time.Date(2023, time.November, 29, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:6, Name:"Chicago Bulls"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12823,
        Date: nba.GameDate{
            Start: time.Date(2023, time.December, 2, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:27, Name:"Philadelphia 76ers"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12853,
        Date: nba.GameDate{
            Start: time.Date(2023, time.December, 13, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:7, Name:"Cleveland Cavaliers"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12867,
        Date: nba.GameDate{
            Start: time.Date(2023, time.December, 15, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:7, Name:"Cleveland Cavaliers"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12877,
        Date: nba.GameDate{
            Start: time.Date(2023, time.December, 16, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:26, Name:"Orlando Magic"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12892,
        Date: nba.GameDate{
            Start: time.Date(2023, time.December, 17, 20, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:26, Name:"Orlando Magic"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12910,
        Date: nba.GameDate{
            Start: time.Date(2023, time.December, 20, 3, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:11, Name:"Golden State Warriors"},
        },
    },
    {
        Id:   12921,
        Date: nba.GameDate{
            Start: time.Date(2023, time.December, 21, 3, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:30, Name:"Sacramento Kings"},
        },
    },
    {
        Id:   12937,
        Date: nba.GameDate{
            Start: time.Date(2023, time.December, 23, 20, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:16, Name:"LA Clippers"},
        },
    },
    {
        Id:   12951,
        Date: nba.GameDate{
            Start: time.Date(2023, time.December, 25, 22, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:17, Name:"Los Angeles Lakers"},
        },
    },
    {
        Id:   12969,
        Date: nba.GameDate{
            Start: time.Date(2023, time.December, 29, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:10, Name:"Detroit Pistons"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12980,
        Date: nba.GameDate{
            Start: time.Date(2023, time.December, 30, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:38, Name:"Toronto Raptors"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   12996,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 1, 0, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:31, Name:"San Antonio Spurs"},
        },
    },
    {
        Id:   13010,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 3, 1, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:25, Name:"Oklahoma City Thunder"},
        },
    },
    {
        Id:   13027,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 6, 0, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:40, Name:"Utah Jazz"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13041,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 7, 0, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:15, Name:"Indiana Pacers"},
        },
    },
    {
        Id:   13055,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 9, 0, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:15, Name:"Indiana Pacers"},
        },
    },
    {
        Id:   13065,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 11, 0, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:22, Name:"Minnesota Timberwolves"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13076,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 12, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:21, Name:"Milwaukee Bucks"},
        },
    },
    {
        Id:   13090,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 14, 0, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:14, Name:"Houston Rockets"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13111,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 16, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:38, Name:"Toronto Raptors"},
        },
    },
    {
        Id:   13119,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 18, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:31, Name:"San Antonio Spurs"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13134,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 20, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:9, Name:"Denver Nuggets"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13151,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 22, 0, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:14, Name:"Houston Rockets"},
        },
    },
    {
        Id:   13159,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 23, 1, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:8, Name:"Dallas Mavericks"},
        },
    },
    {
        Id:   13178,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 26, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:20, Name:"Miami Heat"},
        },
    },
    {
        Id:   13194,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 28, 0, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:16, Name:"LA Clippers"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13207,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 30, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:23, Name:"New Orleans Pelicans"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13218,
        Date: nba.GameDate{
            Start: time.Date(2024, time.January, 31, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:15, Name:"Indiana Pacers"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13232,
        Date: nba.GameDate{
            Start: time.Date(2024, time.February, 2, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:17, Name:"Los Angeles Lakers"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13254,
        Date: nba.GameDate{
            Start: time.Date(2024, time.February, 4, 23, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:19, Name:"Memphis Grizzlies"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13277,
        Date: nba.GameDate{
            Start: time.Date(2024, time.February, 8, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:1, Name:"Atlanta Hawks"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13291,
        Date: nba.GameDate{
            Start: time.Date(2024, time.February, 10, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:41, Name:"Washington Wizards"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13307,
        Date: nba.GameDate{
            Start: time.Date(2024, time.February, 11, 19, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:20, Name:"Miami Heat"},
        },
    },
    {
        Id:   13319,
        Date: nba.GameDate{
            Start: time.Date(2024, time.February, 14, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:4, Name:"Brooklyn Nets"},
        },
    },
    {
        Id:   13328,
        Date: nba.GameDate{
            Start: time.Date(2024, time.February, 15, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:4, Name:"Brooklyn Nets"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13345,
        Date: nba.GameDate{
            Start: time.Date(2024, time.February, 23, 1, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:6, Name:"Chicago Bulls"},
        },
    },
    {
        Id:   13364,
        Date: nba.GameDate{
            Start: time.Date(2024, time.February, 25, 1, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:24, Name:"New York Knicks"},
        },
    },
    {
        Id:   13384,
        Date: nba.GameDate{
            Start: time.Date(2024, time.February, 28, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:27, Name:"Philadelphia 76ers"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13407,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 2, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:8, Name:"Dallas Mavericks"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13420,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 3, 20, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:11, Name:"Golden State Warriors"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13434,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 6, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:7, Name:"Cleveland Cavaliers"},
        },
    },
    {
        Id:   13454,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 8, 3, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:9, Name:"Denver Nuggets"},
        },
    },
    {
        Id:   13468,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 10, 1, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:28, Name:"Phoenix Suns"},
        },
    },
    {
        Id:   13486,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 12, 2, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:29, Name:"Portland Trail Blazers"},
        },
    },
    {
        Id:   13491,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 13, 1, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:40, Name:"Utah Jazz"},
        },
    },
    {
        Id:   13503,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 14, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:28, Name:"Phoenix Suns"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13528,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 17, 22, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:41, Name:"Washington Wizards"},
        },
    },
    {
        Id:   13532,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 18, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:10, Name:"Detroit Pistons"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13546,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 20, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:21, Name:"Milwaukee Bucks"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13559,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 22, 23, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:10, Name:"Detroit Pistons"},
        },
    },
    {
        Id:   13571,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 24, 0, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:6, Name:"Chicago Bulls"},
        },
    },
    {
        Id:   13582,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 25, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:1, Name:"Atlanta Hawks"},
        },
    },
    {
        Id:   13608,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 28, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:1, Name:"Atlanta Hawks"},
        },
    },
    {
        Id:   13622,
        Date: nba.GameDate{
            Start: time.Date(2024, time.March, 30, 21, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:23, Name:"New Orleans Pelicans"},
        },
    },
    {
        Id:   13635,
        Date: nba.GameDate{
            Start: time.Date(2024, time.April, 1, 23, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:5, Name:"Charlotte Hornets"},
        },
    },
    {
        Id:   13653,
        Date: nba.GameDate{
            Start: time.Date(2024, time.April, 3, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:25, Name:"Oklahoma City Thunder"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13666,
        Date: nba.GameDate{
            Start: time.Date(2024, time.April, 5, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:30, Name:"Sacramento Kings"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13682,
        Date: nba.GameDate{
            Start: time.Date(2024, time.April, 7, 22, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:29, Name:"Portland Trail Blazers"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13696,
        Date: nba.GameDate{
            Start: time.Date(2024, time.April, 9, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:21, Name:"Milwaukee Bucks"},
        },
    },
    {
        Id:   13715,
        Date: nba.GameDate{
            Start: time.Date(2024, time.April, 11, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:24, Name:"New York Knicks"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13721,
        Date: nba.GameDate{
            Start: time.Date(2024, time.April, 12, 23, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:5, Name:"Charlotte Hornets"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13734,
        Date: nba.GameDate{
            Start: time.Date(2024, time.April, 14, 17, 0, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:41, Name:"Washington Wizards"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
    {
        Id:   13749,
        Date: nba.GameDate{
            Start: time.Date(2023, time.December, 5, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:2, Name:"Boston Celtics"},
            Home:     nba.Team{Id:15, Name:"Indiana Pacers"},
        },
    },
    {
        Id:   13777,
        Date: nba.GameDate{
            Start: time.Date(2023, time.December, 9, 0, 30, 0, 0, time.UTC),
        },
        Teams: nba.GameTeams{
            Visitors: nba.Team{Id:24, Name:"New York Knicks"},
            Home:     nba.Team{Id:2, Name:"Boston Celtics"},
        },
    },
}
---
